src/Encoding/FilteredStream.php

```diff
@@ -174,15 +179,17 @@ public function isSeekable()
      */
     public function rewind()
     {
-        throw new \RuntimeException('Cannot rewind a filtered stream');
+        @trigger_error('Filtered streams are not seekable. This method will start raising an exception in the next major version');
+        $this->doRewind();
```

Comment:
why is rewind not possible with a filtered stream? is the filter expected to track what it already saw and would need resetting? otherwise it seems to me that only `seek` is problematic, but not `rewind`

Reply:
It really depends on the implementation, but let's say you have a filtered stream that hash the body, rewinding the underlying body would then certainly change the hash produced

Filtered stream act as a pipe, you should never go back.

Also according to PSR7 rewind should raise an exception if the stream is not seekable

Reply:
thanks for the clarification. so basically "you can't know" so you should not do it.

Reply:
exactly